lib/manager/components/reporter/containers/Timetables.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import { connect } from 'react-redux'
+
+import TimetableLayout from '../components/TimetableLayout'
+import { fetchRoutes } from '../../../../gtfs/actions/routes'
+import {
+  timetableDateTimeFilterChange,
+  timetablePatternFilterChange,
+  timetableRouteFilterChange,
+  timetableShowArrivalToggle,
+  timetableTimepointToggle
+} from '../../../../gtfs/actions/timetables'
+import {getTimetableData} from '../../../selectors'
+
+const mapStateToProps = (state, ownProps) => {
+  const {
+    gtfs: {
+      filter,
+      patterns,
+      routes,
+      services,
+      timetables
+    }
+  } = state
+  return {
+    filter,
+    patterns,
+    routes,
+    services,
+    timetables,
+    timetableTableData: getTimetableData(state)
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  const {namespace} = ownProps.version
+  return {
+    onComponentMount: (initialProps) => {
+      if (!initialProps.routes.fetchStatus.fetched) {
+        dispatch(fetchRoutes(namespace))
+      }
+    },
+    routeFilterChange: (newValue) => {
```

Comment:
I thought you had expressed an interest in moving away from longform `mapDispatchToProps` and towards supplying a simple json object. I'm somewhat in favor of that style as well.  Perhaps this container should reflect that as well.

Reply:
I intend to work on that in one big refactor effort as outlined in https://github.com/catalogueglobal/datatools-ui/issues/182.  For now I'm just using whatever the existing coding style is.